rest-api-security/src/main/java/com/epam/learning/backendservices/security/converter/SubscriptionRequestDtoToSubscriptionConverter.java

```diff
@@ -0,0 +1,27 @@
+package com.epam.learning.backendservices.security.converter;
+
+import com.epam.learning.backendservices.security.dto.SubscriptionRequestDto;
+import com.epam.learning.backendservices.security.exeption.PersonNotFoundException;
+import com.epam.learning.backendservices.security.model.Subscription;
+import com.epam.learning.backendservices.security.service.PersonService;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SubscriptionRequestDtoToSubscriptionConverter implements Converter<SubscriptionRequestDto, Subscription> {
+
+    @Autowired
+    private PersonService personService;
+
+    @Override
+    public Subscription convert(SubscriptionRequestDto subscriptionRequestDto) {
+        Subscription subscription = new Subscription();
+        BeanUtils.copyProperties(subscriptionRequestDto, subscription);
+        subscription.setPerson(personService.getPerson(subscriptionRequestDto.getPersonId())
```

Comment:
Be careful when you try to get data from some service inside a mapper. That can result in additional SQL queries, HTTP calls and etc. Optional: come up with better solution.

Reply:
Done